tests/rules/no-missing-label-refs.test.js

```diff
@@ -383,3 +383,12 @@ ruleTester.run("no-missing-label-refs", rule, {
 		},
 	],
 });
+
+it("should not timeout for large inputs", () => {
+	const linter = new Linter();
+	linter.verify(`${"[".repeat(100_000)}x`, {
```

Comment:
On my machine this test passes within ~11 seconds without the fix. Maybe we could increase the input size?
```suggestion
	linter.verify(`${"[".repeat(500_000)}x`, {
```

Reply:
It's true. I guess a more efficient fix would be matching each `[ … ]` substring singularly and then finding those adjacent in the code. That would involve more than changing a regular expression though. If that's a better solution then we should probably do it. We could also leave out the fix for `no-missing-label-refs` in this pull request and do that separately in a follow-up.

Reply:
Not sure if I follow (then again, I'm not very familiar with this code). I'm okay leaving the `no-missing-label-refs` changes out, let me know what you prefer.

Reply:
Ah, I see now that the parser takes up most of the test time. In that case, simply changing the input string might be enough. I tried using `"[abcd".repeat(100_000)` and that cut the test time by over 50% though it still timed out without the fix. There are obviously other options.

Reply:
I see. That test case makes sense, let's roll with it.